scipy/stats/tests/test_multivariate.py

```diff
@@ -1118,15 +1118,21 @@ def test_data_vector_too_long(self):
         assert_raises(ValueError, dirichlet.pdf, x, alpha)
         assert_raises(ValueError, dirichlet.logpdf, x, alpha)
 
-    def test_mean_and_var(self):
+    def test_mean_var_cov(self):
         alpha = np.array([1., 0.8, 0.2])
         d = dirichlet(alpha)
 
-        expected_var = [1. / 12., 0.08, 0.03]
         expected_mean = [0.5, 0.4, 0.1]
+        expected_var = [1. / 12., 0.08, 0.03]
+        expected_cov = [
+                [ 1. / 12, -1. / 15, -1. / 60],
+                [-1. / 15,  2. / 25, -1. / 75],
+                [-1. / 60, -1. / 75,  3. / 100],
+        ]
```

Comment:
Please add a reference (e.g. the code used to compute it in other software) for the expected value. If you wouldn't mind finding a reference for the mean and variance here, while you're at it, I'd appreciate it.

Reply:
It's OK with me to say
> Reference values computed with Mathematica, e.g.
> `Covariance[DirichletDistribution[{ 1, 0.8, 0.2,10^-300}]]`

![image](https://github.com/scipy/scipy/assets/6570539/0db8dce4-180a-4ab7-8442-1244a55abd01)

The 1e-300 is because Mathematica parametrizes differently.

Reply:
I did calculate these values by hand.  I haven't been able to find a third-party reference so far, but I compared the values I calculated to the empirical covariance matrix generated by sampling from the Dirichlet distribution:
```py
from scipy.stats import dirichlet
import numpy as np

alpha = [1, 0.8, 0.2]
d = dirichlet(alpha)
x = d.rvs(size=100000)

empirical_cov = np.cov(x.T)
expected_cov = np.array([
        [ 1. / 12, -1. / 15, -1. / 60],
        [-1. / 15,  2. / 25, -1. / 75],
        [-1. / 60, -1. / 75,  3. / 100],
])

print(f'''\
expected:
{expected_cov}

empirical:
{empirical_cov}

expected - empirical:
{expected_cov - empirical_cov}
''')
```
```
expected:
[[ 0.08333333 -0.06666667 -0.01666667]
 [-0.06666667  0.08       -0.01333333]
 [-0.01666667 -0.01333333  0.03      ]]

empirical:
[[ 0.0834229  -0.06654225 -0.01688065]
 [-0.06654225  0.07975273 -0.01321048]
 [-0.01688065 -0.01321048  0.03009114]]

expected - empirical:
[[-8.95668084e-05 -1.24420657e-04  2.13987465e-04]
 [-1.24420657e-04  2.47269434e-04 -1.22848777e-04]
 [ 2.13987465e-04 -1.22848777e-04 -9.11386881e-05]]
```
I think this shows that the equation is correct.  Of course, this depends on `dirichlet.rvs()` being implemented correctly, but presumably that is accounted for by other tests.  Would it be enough to put a comment like the following in the test?
```
# Reference values calculated by hand, and compared to empirical values determined as follows:
# 
# >>> from scipy.stats import dirichlet
# >>> alpha = [1, 0.8, 0.2]
# >>> d = dirichlet(alpha)
# >>> x = d.rvs(size=100000)
# >>> np.mean(x, axis=0)
# array([0.49900016, 0.40103127, 0.09996857])
# >>> np.cov(x.T)
# array([[ 0.08316152, -0.06662866, -0.01653286],
#        [-0.06662866,  0.08002796, -0.0133993 ],
#        [-0.01653286, -0.0133993 ,  0.02993216]])
```

Reply:
Yes, we've used that sort of strategy for tests. However, usually, we'd use that to test the `rvs` method, with more precise tests of the covariance. It's unlikely that they'd agree unless both were correct, though.

If the suggestion https://github.com/scipy/scipy/pull/18664#discussion_r1227295365 looks good to you, I can just commit that. Once tests pass, I can merge this.

Reply:
Yeah, the Mathematica suggestion looks good to me.  I'd probably write that the values were calculated by hand and compared to the Mathematica values, just because it's clear that Mathematica wouldn't output whole-number fractions.  But it doesn't really matter.  Thanks for doing some legwork on this for me!

Reply:
Yeah, I had something like that but simplified it. But yeah, I'll change it back.